packages/marketplace/src/components/ui/analytics/billing/cost-calculator/total-cost-table.tsx

```diff
@@ -1,101 +1,110 @@
-    const { limit, price } = tierPriceLimit
-    bandCostPerCall.push(price)
-
-    if (remainingApiCalls > limit) {
-      bands.push(limit)
-      totalMonthlyCost += limit * price
-      remainingApiCalls -= limit
-    } else {
-      bands.push(remainingApiCalls)
-      totalMonthlyCost += remainingApiCalls * price
-      remainingApiCalls = 0
-      break
+export const prepareData = (endpointsUsed: string, apiCalls: string, foundationPricing: TierPrice) => {
+  return (): TotalCostTableData => {
+    const bands: number[] = []
+    const bandCostPerCall: number[] = []
+    let totalMonthlyCost: number = 0
+
+    let remainingApiCalls = parseFloat(apiCalls)
+    const { priceRange, maxPrice } = foundationPricing[endpointsUsed]
+
+    for (const tierPriceLimit of priceRange) {
+      const { limit, price } = tierPriceLimit
+      bandCostPerCall.push(price)
+
+      if (remainingApiCalls > limit) {
+        bands.push(limit)
+        totalMonthlyCost += limit * price
+        remainingApiCalls -= limit
+      } else {
+        bands.push(remainingApiCalls)
+        totalMonthlyCost += remainingApiCalls * price
+        remainingApiCalls = 0
+        break
+      }
     }
-  }
-
-  if (remainingApiCalls > 0) {
-    bands.push(remainingApiCalls)
-    bandCostPerCall.push(maxPrice)
-    totalMonthlyCost += maxPrice * remainingApiCalls
-  }
-
-  const costPerBand = bandCostPerCall.map((item, index) => {
-    return item * bands[index]
-  })
-
-  return {
-    bands,
-    bandCostPerCall,
-    costPerBand,
-    totalMonthlyCost,
-  }
-}
 
-export const renderEndpointsUsed = (bands, endpointsUsedRange, endpointsUsed) => {
-  return bands.map((_: any, index) => {
-    if (index === 0) {
-      return <td key={index}>{endpointsUsedRange[endpointsUsed]}</td>
+    if (remainingApiCalls > 0) {
+      bands.push(remainingApiCalls)
+      bandCostPerCall.push(maxPrice)
+      totalMonthlyCost += maxPrice * remainingApiCalls
     }
-    return <td key={index}></td>
-  })
-}
 
-export const renderApiCalls = (bands, apiCalls) => {
-  return bands.map((_: any, index) => {
-    if (index === 0) {
-      return <td key={index}>{formatNumber(parseFloat(apiCalls))}</td>
+    const costPerBand = bandCostPerCall.map((item, index) => {
+      return item * bands[index]
+    })
+
+    const tableData = bands.map((item, index) => {
+      return {
+        numberOfApiCalls: item,
+        costPerApiCall: bandCostPerCall[index],
+        totalCost: costPerBand[index],
+      }
+    })
+    return {
+      tableData,
+      totalMonthlyCost,
     }
-    return <td key={index}></td>
-  })
-}
-
-export const renderBand = bands => {
-  return bands.map((item, index) => {
-    return <td key={index}>{formatNumber(item)}</td>
-  })
+  }
 }
 
-export const renderBandCostPerCall = bandCostPerCall => {
-  return bandCostPerCall.map((item, index) => {
-    return <td key={index}>{formatCurrency(item, 6)}</td>
-  })
+export const prepareTableColumns = totalMonthlyCost => {
```

Comment:
What is totalMonthly cost? Can we avoid implicit any please by adding definitions, thanks